src/modules/ProductPage/ProductPage.tsx

```diff
@@ -0,0 +1,310 @@
+import { Breadcrumbs } from '../../components/Breadcrumbs';
+import { ProductsSlider } from '../../components/ProductsSlider';
+import { HeadingLevel } from '../../types/HeadingLevel';
+import { Title } from '../../components/Title';
+import { ProductSlider } from '../../components/ProductSlider';
+import { Product } from '../../types/Product';
+import { Link, useLocation, useParams } from 'react-router-dom';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import classNames from 'classnames';
+import { cartSlice } from '../../features/cart';
+import { favouriteSlice } from '../../features/favourite';
+import { BackButton } from '../../components/BackButton';
+import { useEffect, useState } from 'react';
+import {
+  getAccessories,
+  getAllProducts,
+  getPhones,
+  getTablets,
+} from '../../api/products';
+import { ProductData } from '../../types/ProductData';
+
+export const ProductPage = () => {
+  const [product, setProduct] = useState<ProductData | null>(null);
+  const [products, setProducts] = useState<Product[]>([]);
+  const [categoryProducts, setCategoryProducts] = useState<ProductData[]>([]);
+  const dispatch = useAppDispatch();
+  const cart = useAppSelector(state => state.cart);
+  const favourite = useAppSelector(state => state.favourite);
+  const { itemId } = useParams();
+  const location = useLocation();
+  const category = location.pathname.slice(1).split('/')[0];
+  const isInCart = cart.find((item: Product) => item.itemId === product?.id);
+  const isFavourite = favourite.find(
+    (item: Product) => item.itemId === product?.id,
+  );
+  const mayLike = products.filter(
+    (item: Product) => item.category === product?.category,
+  );
```

Comment:
it would be nice to create more personalized recommendations based on attributes like namespaceId, color or other features of selected item because currently the same array of products is shown for every item in the same category including selected item itself ^^"